.circleci/config.yml

```diff
@@ -91,40 +85,42 @@ jobs:
       - store_artifacts:
           path: /nerves/deploy/system/artifacts
           destination: system
-      - save_cache:
-          key: nerves/build-{{ .Revision }}-{{ .Environment.CIRCLE_TAG }}
-          paths:
-            - "/nerves/build"
       - save_cache:
           key: nerves/deploy/system-{{ .Revision }}-{{ .Environment.CIRCLE_TAG }}
           paths:
             - "/nerves/deploy/system"
 
   build_test:
     <<: *defaults
-    working_directory: /nerves/build/test
     steps:
+      - checkout
       - restore_cache:
-          key: nerves/build-{{ .Revision }}-{{ .Environment.CIRCLE_TAG }}
+          key: nerves/deploy/system-{{ .Revision }}-{{ .Environment.CIRCLE_TAG }}
+      - run:
+          name: Copy artifacts
+          command: |
+            mkdir -p ~/.nerves/dl
+            cp /nerves/deploy/system/artifacts/*.gz ~/.nerves/dl
```

Comment:
What's this part for?

Reply:
There needs to be a mechanism to share the built artifacts between job containers. Before this PR we would package up the entire contents of the build directory and then restore it in the next container. This would take a couple minutes and was unnecessary. We already have a cache going for the archived artifact. Prior to this step we restore the system artifact cache and then copy the archive out into the `NERVES_DL_DIR`. This is similar to how we prime things for training. When the test app goes to look for the artifact, it will see that it has one downloaded already and decompress it.

Reply:
Got it. I forgot that there was a multi-part process going on with the CircleCI configs and the build products needed to be stashed somewhere between steps.